README.md

```diff
@@ -41,31 +41,100 @@ callback parameter like so:
       }
     });
 
-The `eachLine` function also returns an object with one property, `then`.  If a
-callback is provided to `then`, it will be called once all lines have been read.
+You can provide an optional second node-style callback that will be called with
+`(err)` on failure or `()` when finished (even if you manually terminate iteration
+by returning `false` from the iteratee):
 
     var lineReader = require('line-reader');
 
     // read all lines:
     lineReader.eachLine('file.txt', function(line) {
       console.log(line);
-    }).then(function () {
+    }).then(function (err) {
+      if (err) throw err;
       console.log("I'm done!!");
     });
 
-
 For more granular control, `open`, `hasNextLine`, and `nextLine` maybe be used
-to iterate a file:
+to iterate a file (but you must `close` it yourself):
```

Comment:
I could definitely provide an example, though it might not be an idiom reusable in all cases (since there could be many possible code paths).  I could make the library autoclose the file on error or eof (`fs.createReadStream` has such an option), but it's not possible to autoclose the file when the user arbitrarily decides to stop reading lines.  What do you think?